06x-new-kafka-roller.md

```diff
@@ -0,0 +1,331 @@
+# Kafka Roller 2.0
+
+## Current situation
+
+The Kafka Roller is an internal Cluster Operator component that's responsible for coordinating the rolling restart or reconfiguration of Kafka pods when:
+- non-dynamic reconfigurations needs to be applied
+- update in Kafka CRD is detected
+- a certificate is renewed
+- pods have been manually annotated by the user for controlled restarts
+- pod is stuck and is out of date
+- Kafka broker is unresponsive to Kafka Admin connections
+
+A pod is considered stuck if it is in one of following states:
+- `CrashLoopBackOff`
+- `ImagePullBackOff`
+- `ContainerCreating`
+- `Pending` and `Unschedulable`
```

Comment:
Currently if we find that the pod is in one of these states and does not have old revision, we fail the reconciliation and it will be checked again in the next  reconciliation. We don't specifically check if it's in a state more than X seconds.